src/java.base/share/classes/jdk/internal/value/PrimitiveClass.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 1994, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.value;
+
+import jdk.internal.access.JavaLangAccess;
+import jdk.internal.access.SharedSecrets;
+import jdk.internal.vm.annotation.IntrinsicCandidate;
+
+/**
+ * Utilities to access Primitive Classes as described in JEP 401.
+ */
+public class PrimitiveClass {
+
+    // ACC_PRIMITIVE from JEP 401
+    private static final int PRIMITIVE_CLASS = 0x00000800;
+
+    private static final JavaLangAccess javaLangAccess = SharedSecrets.getJavaLangAccess();
+
+
+    /**
+     * Returns a {@code Class} object representing the primary type
+     * of this class or interface.
+     * <p>
+     * If this {@code Class} object represents a primitive type or an array type,
+     * then this method returns this class.
+     * <p>
+     * If this {@code Class} object represents a {@linkplain #isPrimitiveClass(Class)
+     * primitive class}, then this method returns the <em>primitive reference type</em>
+     * type of this primitive class.
+     * <p>
+     * Otherwise, this {@code Class} object represents a non-primitive class or interface
+     * and this method returns this class.
+     *
+     * @param aClass a class
+     * @return the {@code Class} representing the primary type of
+     *         this class or interface
+     * @since Valhalla
+     */
+    @IntrinsicCandidate
```

Comment:
The compiler intrinsic is for the method in `java.lang.Class`.  So the annotation should be dropped.  Same for `asValueType`.